bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/GoveeDiscoveryService.java

```diff
@@ -0,0 +1,265 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.binding.govee.internal;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.MulticastSocket;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.govee.internal.model.DiscoveryResponse;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.i18n.TranslationProvider;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Discovers Govee Devices
+ *
+ * Scan approach:
+ * 1. Determines all local network interfaces
+ * 2. Send a multicast message on each interface to the Govee multicast address 239.255.255.250 at port 4001
+ * 3. Retrieve the list of devices
+ *
+ * Based on the description at https://app-h5.govee.com/user-manual/wlan-guide
+ *
+ * A typical scan response looks as follows
+ *
+ * {
+ * "msg":{
+ * "cmd":"scan",
+ * "data":{
+ * "ip":"192.168.1.23",
+ * "device":"1F:80:C5:32:32:36:72:4E",
+ * "sku":"Hxxxx",
+ * "bleVersionHard":"3.01.01",
+ * "bleVersionSoft":"1.03.01",
+ * "wifiVersionHard":"1.00.10",
+ * "wifiVersionSoft":"1.02.03"
+ * }
+ * }
+ * }
```

Comment:
I'm not sure about the syntax, but I believe the `@code` attribute should be applied here.

Reply:
Yep, that seems to work. Changed.